src/runtime/scheduler_legacy.go

```diff
@@ -0,0 +1,100 @@
+package runtime
+
+import "internal/task"
+
+var (
+	sleepQueue         *task.Task
+	sleepQueueBaseTime timeUnit
+)
+
+// Add this task to the sleep queue, assuming its state is set to sleeping.
+func addSleepTask(t *task.Task, duration int64) {
+	if schedulerDebug {
+		println("  set sleep:", t, uint(duration/tickMicros))
+		if t.Next != nil {
+			panic("runtime: addSleepTask: expected next task to be nil")
+		}
+	}
+	t.Data = uint(duration / tickMicros) // TODO: longer durations
+	now := ticks()
+	if sleepQueue == nil {
+		scheduleLog("  -> sleep new queue")
+
+		// set new base time
+		sleepQueueBaseTime = now
+	}
+
+	// Add to sleep queue.
+	q := &sleepQueue
+	for ; *q != nil; q = &(*q).Next {
+		if t.Data < (*q).Data {
+			// this will finish earlier than the next - insert here
+			break
+		} else {
+			// this will finish later - adjust delay
+			t.Data -= (*q).Data
+		}
+	}
+	if *q != nil {
+		// cut delay time between this sleep task and the next
+		(*q).Data -= t.Data
+	}
+	t.Next = *q
+	*q = t
+}
+
+// pollSleepQueue checks the sleep queue to see if any tasks are now ready to run.
+func pollSleepQueue(now timeUnit) bool {
+	// Add tasks that are done sleeping to the end of the runqueue so they
+	// will be executed soon.
+	var awoke bool
+	for sleepQueue != nil && now-sleepQueueBaseTime >= timeUnit(sleepQueue.Data) {
+		t := sleepQueue
+		scheduleLogTask("  awake:", t)
+		sleepQueueBaseTime += timeUnit(t.Data)
+		sleepQueue = t.Next
+		t.Next = nil
+		runqueue.Push(t)
+		awoke = true
+	}
+
+	return awoke
+}
+
+var curTime timeUnit
```

Comment:
Suggestion: I personally prefer to put global variables and consts at the top of the file, to make them more visible. I'm not aware of any official guidelines in this regard but it's something which I often see in code (including Go code).

Reply:
I tend to put variables closer to their use, as this makes it easier to track what they are used for.